ingest/de.py

```diff
@@ -38,36 +39,54 @@ def __init__(
         annotation,
         **kwargs,
     ):
+        DifferentialExpression.de_logger.info(
+            "Initializing DifferentialExpression instance"
+        )
         self.cluster = cluster
         self.metadata = cell_metadata
         self.annotation = annotation
         self.matrix_file_path = matrix_file_path
         self.matrix_file_type = matrix_file_type
         self.kwargs = kwargs
         self.accession = self.kwargs["study_accession"]
-        # only used in output filename, removing spaces
-        self.cluster_name = self.kwargs["name"].replace(" ", "_")
+        # only used in output filename, replacing non-alphanumeric with underscores
+        self.cluster_name = re.sub(r'\W+', '_', self.kwargs["name"])
         self.method = self.kwargs["method"]
 
         if matrix_file_type == "mtx":
             self.genes_path = self.kwargs["gene_file"]
-            genes_ingest_file = IngestFiles(self.genes_path, self.ALLOWED_FILE_TYPES)
-            self.genes_file = genes_ingest_file.resolve_path(self.genes_path)[0]
-            self.genes: List[str] = [
-                g.strip().strip('"') for g in self.genes_file.readlines()
-            ]
-
             self.barcodes_path = self.kwargs["barcode_file"]
-            barcodes_ingest_file = IngestFiles(
-                self.barcodes_path, self.ALLOWED_FILE_TYPES
+
+    @staticmethod
+    def assess_annotation(annotation, metadata):
+        """ Check that annotation for DE is not of TYPE numeric
+        """
+        dtype_info = dict(zip(metadata.headers, metadata.annot_types))
+        annotation_info = dtype_info.get(annotation, "not found")
```

Comment:
I would use `None` instead of "not found" as the default here -- no reason to make a 'magic' string